src/renderer/cluster-frame.tsx

```diff
@@ -103,7 +103,8 @@ export class ClusterFrame extends React.Component {
 
     // Only load the extensions once the catalog has been populated
     when(
-      () => Boolean(catalogEntityRegistry.activeEntity),
+      // watch for .items, as .activeEntity might be populated only after extensions are loaded (if using custom Catalog Category)
```

Comment:
This sounds a bit wrong, shouldn't `.activeEntity` work also in this case?

Reply:
Especially since we still need `activeEntity` to be truthy for the next line.

Reply:
activeEntity doesn't work if the entity happens to be of custom type, e.g. in my case ManagedDevCluster, and that type is added by an extension. In that case `getEntityForData` returns null, because it can't find the category. But we load the extensions only after the when resolves, and that will add the custom category.

@Nokel81's point seems correct though, not sure why the UI still seems to work. Probably I need to refactor this to have 3 steps: 
1. wait for catalog items
2. load extensions
3. Do the stuff in `loadOnClusterRenderer` with a known `entity`

Reply:
Improved this part. It still a bit.. dangerous because of this; we can have 10 extensions, only one of which will register the new class, so we should first  load the correct extension, and then the others that might use isEnabledForCluster.

Reply:
Improved more, should be fixed now.